tests/test_utils.py

```diff
@@ -45,11 +47,18 @@ def test_get_total_number_of_trainable_parameters():
 
     # Calculate the expected number of trainable parameters
     expected_params = sum(p.numel() for p in model.parameters() if p.requires_grad)
```

Comment:
```suggestion
    expected_params = 10*5 + 5 + 5*2 + 2  # weights_1 + bias_1 + weights_2 + bias_2 = 67
```
I would replace the automated parameter counting by manual parameter counting. Not only here, but especially in line 38 as well. The test `test_get_local_number_of_trainable_parameters` seems a bit meaningless otherwise, since it literally uses the same line of code as the tested function (Note: the problem was not introduced by this PR)